app/jinja_filters.py

```diff
@@ -43,6 +43,23 @@ def format_date(value):
     return value.strftime('%-d %B %Y')
 
 
+@blueprint.app_template_filter()
+def format_conditional_date(dates):
+    """ This function format_cond_date accepts two dates, if a date is submitted that is used and is available,
+    if not the metadata date is made available The format (example) in the json is
+    {{[answers.date_from, exercise.start_date]|format_conditional_date}} """
+
+    if dates[0]:
+        date = dates[0]
+    else:
+        date = dates[1]
+
```

Comment:
@berianjones line 58 would fail with `IndexError: list index out of range` so would not get to line 60 to be checked.